lib/msal-browser/test/cache/BrowserCacheManager.spec.ts

```diff
@@ -325,7 +326,532 @@ describe("BrowserCacheManager tests", () => {
+                    expect(browserLocalStorage.getServerTelemetry(key)).toBeNull();
+                });
+
+                it("getThrottlingCache returns null if value is not JSON", () => {
+                    const key = "testKey";
+                    browserLocalStorage.setItem(key, "this is not json");
+                    browserSessionStorage.setItem(key, "this is not json");
+
+                    expect(browserSessionStorage.getThrottlingCache(key)).toBeNull();
+                    expect(browserLocalStorage.getThrottlingCache(key)).toBeNull();
+                });
+
+                it(
+                    "getThrottlingCache returns null if value is not throttling entity",
+                    () => {
+                        const key = "testKey";
+                        const partialThrottlingEntity = {
+                            error: "error"
+                        };
+
+                        browserLocalStorage.setItem(key, JSON.stringify(partialThrottlingEntity));
+                        browserSessionStorage.setItem(key, JSON.stringify(partialThrottlingEntity));
+
+                        expect(browserSessionStorage.getThrottlingCache(key)).toBeNull();
+                        expect(browserLocalStorage.getThrottlingCache(key)).toBeNull();
+                    }
+                );
+
+                it("getThrottlingCache returns ThrottlingEntity", () => {
+                    const testKey = "throttling";
+                    const testVal = new ThrottlingEntity();
+                    testVal.throttleTime = 60;
+
+                    browserLocalStorage.setThrottlingCache(testKey, testVal);
+                    browserSessionStorage.setThrottlingCache(testKey, testVal);
+
+                    expect(browserSessionStorage.getThrottlingCache(testKey)).toEqual(testVal);
+                    expect(browserSessionStorage.getThrottlingCache(testKey)).toBeInstanceOf(ThrottlingEntity);
+                    expect(browserLocalStorage.getThrottlingCache(testKey)).toEqual(testVal);
+                    expect(browserLocalStorage.getThrottlingCache(testKey)).toBeInstanceOf(ThrottlingEntity);
+                });
+            });
+
+            describe("RedirectRequestContext", () => {
+                it("Returns redirect request context as null if context not set in browser cache", () => {
+                    expect(browserSessionStorage.getRedirectRequestContext()).toEqual(null);
+                });
+
+                it("Returns redirect request context if context set in browser cache", () => {
+                    const testVal = "testId";
+                    browserSessionStorage.setRedirectRequestContext(testVal);
+                    expect(browserSessionStorage.getRedirectRequestContext()).toEqual(testVal);
+                });
+
+            });
+        });
+    });
+
+    
+    describe("Interface functions with overridden temporaryCacheLocation", () => {
```

Comment:
Test additions above this line don't appear relevant to this PR

Reply:
I have copy-pasted these tests from "Interface functions" tests. But in this test setup I changed the cacheConfig for the temporaryCacheLocation to LocalStorage. The reasoning behind this was to show that the existing functionality doesn't break with the temporaryCacheLocation to LocalStorage.

Reply:
If you prefer having them removed, let me know

Reply:
These tests were already there

Reply:
These tests were already there. I just moved them in this file in a more logical order